src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/MemberExpansion.cs

```diff
@@ -38,9 +38,9 @@ internal sealed class MemberExpansion : Expansion
             }
 
             var runtimeType = type.GetLmrType();
-            // Primitives, enums and null values with a declared type that is an interface have no visible members.
+            // Primitives, enums, function pointers, and null values with a declared type that is an interface have no visible members.
             Debug.Assert(!runtimeType.IsInterface || value.IsNull);
-            if (formatter.IsPredefinedType(runtimeType) || runtimeType.IsEnum || runtimeType.IsInterface)
+            if (formatter.IsPredefinedType(runtimeType) || runtimeType.IsEnum || runtimeType.IsInterface || runtimeType.IsFunctionPointer())
```

Comment:
I tested it against a binary built by the managed C++ compiler.  I found these two spots by simply expanding an object with a function pointer member and debugging the crash.  I tried to the repro again with the fix and confirmed that it worked.